livekit/src/room/mod.rs

```diff
@@ -783,6 +783,7 @@ impl RoomSession {
             let remote_participant = self.get_participant_by_sid(&participant_sid);
             if pi.state == proto::participant_info::State::Disconnected as i32 {
                 if let Some(remote_participant) = remote_participant {
+                    remote_participant.update_info(pi.clone());
```

Comment:
even when the remote participant is disconnected, we have to update it to keep its reason